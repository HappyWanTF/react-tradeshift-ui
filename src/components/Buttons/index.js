import PropTypes from 'prop-types';
import React from 'react';
import ShortID from 'shortid';

const Buttons = ({ children, ...props }) => {
	return (
		<menu data-ts="Buttons" {...props}>
			{children.map(child => (
				<li key={ShortID.generate()}>{child}</li>
			))}
		</menu>
	);
};

Buttons.propTypes = {
	children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Buttons.defaultProps = {
	children: null
};

export default Buttons;
