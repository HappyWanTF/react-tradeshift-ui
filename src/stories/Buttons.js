import { storiesOf } from '@storybook/react';
import React from 'react';
import { Buttons } from '../components';

const stories = storiesOf('Buttons', module);

stories.add(
	'Basic structure',
	() => (
		<Buttons>
			<button type="button">
				<span>Button One</span>
			</button>
			<button type="button">
				<span>Button Two</span>
			</button>
		</Buttons>
	),
	{
		info: `The basic structure`
	}
);

stories.add(
	'Joined buttons',
	() => (
		<Buttons className="ts-t-t ts-join">
			<button type="button">
				<span>Button One</span>
			</button>
			<button type="button">
				<span>Button Two</span>
			</button>
		</Buttons>
	),
	{
		info:
			'Buttons component accepts attributes describes in `tradeshift-ui buttons` documentation page'
	}
);
