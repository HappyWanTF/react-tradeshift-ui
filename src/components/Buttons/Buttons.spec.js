import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ShortID from 'shortid';
import Buttons from '.';

configure({ adapter: new Adapter() });

describe('Buttons', () => {
	beforeEach(() => {
		jest
			.spyOn(ShortID, 'generate')
			.mockReturnValueOnce('short_id_1')
			.mockReturnValueOnce('short_id_2')
			.mockReturnValueOnce('short_id_3')
			.mockReturnValueOnce('short_id_4');
	});

	it('should render correctly', () => {
		const wrapper = shallow(
			<Buttons>
				<button type="button">
					<span>Button One</span>
				</button>
				<button type="button">
					<span>Button Two</span>
				</button>
			</Buttons>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
