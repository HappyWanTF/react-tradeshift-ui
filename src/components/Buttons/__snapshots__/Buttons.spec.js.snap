// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render correctly 1`] = `
<menu
  data-ts="Buttons"
>
  <li
    key="short_id_1"
  >
    <button
      type="button"
    >
      <span>
        Button One
      </span>
    </button>
  </li>
  <li
    key="short_id_2"
  >
    <button
      type="button"
    >
      <span>
        Button Two
      </span>
    </button>
  </li>
</menu>
`;
